### ISSUE TITLE
Unhelpful CS0119 error when unable to find a static abstract method to resolve

### ISSUE THREAD

**Issue by AlFasGD** [dotnet/roslyn#61486]
**Version Used**: 17.3.0 Preview 1.1

**Steps to Reproduce**:

1. Declare a method that has access to a type parameter (either from the containing type, or the method itself)
2. Try to use an unavailable static abstract method for the given type parameter

**Expected Behavior**: A specific error about the unavailability of the method.

**Actual Behavior**: An unhelpful CS0119 error, without hinting the availability of a feature enabling calling static abstract methods from type parameter arguments.

**Screenshots**:
![image](https://user-images.githubusercontent.com/8298332/170092974-7da9c8a7-f6d8-4649-a112-b22d56464368.png)

**Comment by jcouv** [comment]
I agree the diagnostic seems unhelpful/misleading:
```
public interface I 
{
    //static abstract void Method();
}

public class C<T> where T : I
{
    public void M() 
    {
        T.Method(); // error CS0119: 'T' is a type parameter, which is not valid in the given context
    }
}
```

@AlekseyTs Do you have an idea how to improve this experience?

**Comment by AlekseyTs** [comment]
> Do you have an idea how to improve this experience?

Adding more info to the message is an option: "'T' is a type parameter, which is not valid in the given context **, unless to access a static virtual or abstract member**"
